src/Services/UpdateRecords/enumerateData.service.ts

```diff
@@ -0,0 +1,19 @@
+type UpdateOrCreateFunction<RecordType, CompareType> = (
+    comparisonObject: any,
+    record: RecordType,
+    compareValues: CompareType
+) => Promise<void>;
+
+export const enumerateData = <RecordType, CompareType>(
+    records: RecordType[],
+    compareValues: CompareType,
+    updateOrCreateData: UpdateOrCreateFunction<RecordType, CompareType>,
+    generateComparisonObject: (record: RecordType) => any
+) => {
+    if (records && records.length > 0) {
+        records.forEach(async (record: RecordType) => {
+            const comparisonObject = generateComparisonObject(record);
+            await updateOrCreateData(comparisonObject, record, compareValues);
+        });
+    }
+};
```

Comment:
```suggestion
export const enumerateData = <RecordType, CompareType>(
    records: RecordType[],
    compareValues: CompareType,
    updateOrCreateData: UpdateOrCreateFunction<RecordType, CompareType>,
    comparisionKey: key of RecordType
) => {
    if (records && records.length > 0) {
        records.forEach(async (record: RecordType) => {
            const comparisonObject = record[comparisionKey]
            await updateOrCreateData(comparisonObject, record, compareValues);
        });
    }
};
```

Reply:
not sure if this is a better way in this case